frontend/src/components/main/Main.js

```diff
@@ -87,7 +98,16 @@ export default function Main(props) {
             className="address-input"
             disabled={ props.fetching }
             value={input2}
-            onChange={ (e) => setInput2(e.target.value) }
+            onChange={ (e) => {
+              setInput2(e.target.value)
+              if(vaultStore.balanceOf.value !== 0){
+                setInput1(e.target.value / tokenStore.ratioToken);
+              }
+              if(tokenStore.balanceDai < parseFloat(e.target.value) * Math.pow(10, tokenStore.decimalsDai)) {
```

Comment:
this if block its very similar to the previous one, let abstract it into a function internal to this component?
Also it would be helpful to not cause a loop of changes, because if we have approximation errors this will create a loop of `onChange` that can go to infinite

Reply:
Good idea to use a function internal,
No, onChange listen only to input event, so it won't cause infinite loop